app/controllers/reputation_web_service_controller.rb

```diff
@@ -163,87 +250,32 @@ def send_post_request
-        end
-      when '735' # expert grades of program 1 (735)
-        req.body.prepend('"expert_grades": {"submission24083":96.084,"submission24085":88.811,"submission24086":100,"submission24087":100,"submission24088":92.657,"submission24091":96.783,"submission24092":90.21,"submission24093":100,"submission24097":90.909,"submission24098":98.601,"submission24101":99.301,"submission24278":98.601,"submission24279":72.727,"submission24281":54.476,"submission24289":94.406,"submission24291":99.301,"submission24293":93.706,"submission24296":98.601,"submission24302":83.217,"submission24303":91.329,"submission24305":100,"submission24307":100,"submission24308":100,"submission24311":95.804,"submission24313":91.049,"submission24314":100,"submission24315":97.483,"submission24316":91.608,"submission24317":98.182,"submission24320":90.21,"submission24321":90.21,"submission24322":98.601},')
-      when '754' # expert grades of Wiki contribution (754)
-        req.body.prepend('"expert_grades": {"submission25030":95,"submission25031":92,"submission25033":88,"submission25034":98,"submission25035":100,"submission25037":95,"submission25038":95,"submission25039":93,"submission25040":96,"submission25041":90,"submission25042":100,"submission25046":95,"submission25049":90,"submission25050":88,"submission25053":91,"submission25054":96,"submission25055":94,"submission25059":96,"submission25071":85,"submission25082":100,"submission25086":95,"submission25097":90,"submission25098":85,"submission25102":97,"submission25103":94,"submission25105":98,"submission25114":95,"submission25115":94},')
-      when '756' # expert grades of Wikipedia contribution (756)
-        req.body.prepend('"expert_grades": {"submission25107":76.6667,"submission25109":83.3333},')
-        # comment end
-      end
+      req.body = add_expert_grades(req.body)
     elsif params[:checkbox][:hamer] == 'Add initial Hamer reputation values'
-      @@additional_info = 'add initial hamer reputation values'
+      add_hamer_reputation_values
     elsif params[:checkbox][:lauw] == 'Add initial Lauw reputation values'
-      @@additional_info = 'add initial lauw reputation values'
+      add_lauw_reputation_values
     elsif params[:checkbox][:quiz] == 'Add quiz scores'
-      @@additional_info = 'add quiz scores'
-      assignment_id_list_quiz = get_assignment_id_list(params[:assignment_id].to_i, params[:another_assignment_id].to_i)
-      quiz_str =  generate_json_for_quiz_scores(assignment_id_list_quiz).to_json
-      # quiz_str = json_generator(params[:assignment_id].to_i, params[:another_assignment_id].to_i, params[:round_num].to_i, 'quiz scores').to_json
-      quiz_str[0] = ''
-      quiz_str.prepend('"quiz_scores":{')
-      quiz_str += ','
-      quiz_str = quiz_str.gsub('"N/A"', '20.0')
-      req.body.prepend(quiz_str)
+      req.body = add_quiz_scores(req.body)
     else
       @@additional_info = ''
```

Comment:
can you create a separate method to set additional_info in a method in a dry fashion? also we might need to eliminate the use of class variables here, as it is not good to pass information around using class variables. we need to use function parameters to pass info, we can do that in the next PullRequest though, no need to change anything here.

Reply:
Will deal with it in the next pr